src/tls_certificate_check.erl

```diff
@@ -103,6 +105,11 @@ options(Target) ->
             []
     end.
 
+%% @doc Blocks until shared state is initialized.
+-spec ensure_state_initialized() -> ok | {error, timeout}.
+ensure_state_initialized() ->
```

Comment:
I guess this function could either be just exposed to the user so it can be called optionally before calling `options`, not exported but called inside `options`, or both?